src/sharedInterfaces/connectionDialog.ts

```diff
@@ -122,11 +129,11 @@ export interface FormComponentOptions {
 /**
  * Interface for a form event
  */
-export interface FormEvent {
+export interface FormEvent<T> {
```

Comment:
Pull `FormContextProps<T>`, `FormEvent<T>`, and baseclass for `ConnectionDialogReducers` to Form-dedicated file